go/internal/feast/errors.go

```diff
@@ -0,0 +1,21 @@
+package feast
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TODO: Add more errors here especially once that have been created in python sdk
+// and coordinate error exceptions in GoServer in python sdk
+type FeatureNameCollisionError struct {
+	featureRefCollisions []string
+	fullFeatureNames     bool
+}
+
+func NewFeatureNameCollisionError(featureRefCollisions []string, fullFeatureNames bool) FeatureNameCollisionError {
+	return FeatureNameCollisionError{featureRefCollisions, fullFeatureNames}
+}
+
+func (e FeatureNameCollisionError) Error() string {
+	return fmt.Sprintf("featureNameCollisionError: %s; %t", strings.Join(e.featureRefCollisions, ", "), e.fullFeatureNames)
```

Comment:
Should we use plain english for the error message instead of an error name?

Reply:
This is needed to pass one of the integration tests that requires FeatureNameCollision exception from python sdk. I hacked around by parsing error message to coordinate between 2 languages but would there be a better coordination method I should use instead?

Reply:
I just want to prevent polluting the Golang SDK with Python specific requirements. At the very least a translation layer would be useful here to go from a Go error message to this Python one, even if it's hardcoded. It may also be a good idea to eventually introduce error codes instead of having to parse strings.